plugins/hfp_ag_bluez5.c

```diff
@@ -178,11 +232,11 @@ static DBusMessage *profile_disconnection(DBusConnection *conn,
 
 	DBG("%s", device);
 
-	fd = g_hash_table_lookup(connection_hash, device);
-	if (fd == NULL)
+	dev = g_hash_table_lookup(connection_hash, device);
+	if (!dev)
 		goto invalid;
 
-	shutdown(GPOINTER_TO_INT(fd), SHUT_RDWR);
+	shutdown(GPOINTER_TO_INT(dev->fd), SHUT_RDWR);
```

Comment:
GPOINTER_TO_INT is not needed here, fd is not a pointer any more (now we store dev in the hash table instead).